extensions/amp-story/0.1/amp-story.js

```diff
@@ -917,6 +895,8 @@ export class AmpStory extends AMP.BaseElement {
     }
 
     this.buildBookend_().then(() => {
+      this.navigationState_.fire(StateChangeType.BOOKEND_ENTER);
```

Comment:
IMO this breaks `NavigationState`'s encapsulation. Can we keep `fire()` private and instead have it detect when the bookend is entered/exited?

Reply:
Good idea.